### PR TITLE
Add TargetPointMultiplier

### PR DESCRIPTION
**Pull request by isuruf**
To reproduce the results of https://doi.org/10.1016/j.jcp.2007.06.029

- [ ] Review https://github.com/inducer/pytential/pull/75 and merge at the same time

### CODE REVIEW

File: sumpy/e2p.py [hunk 1]
```diff
@@ -56,9 +56,10 @@ def __init__(self, ctx, expansion, kernels,
         if device is None:
             device = ctx.devices[0]
 
-        from sumpy.kernel import SourceDerivativeRemover, TargetDerivativeRemover
-        sdr = SourceDerivativeRemover()
-        tdr = TargetDerivativeRemover()
+        from sumpy.kernel import (SourceTransformationRemover,
+                TargetTransformationRemover)
+        sdr = SourceTransformationRemover()
+        tdr = TargetTransformationRemover()
```
Comment (inducer) [inline sumpy/e2p.py:62]:
```suggestion
        sxr = SourceTransformationRemover()
        txr = TargetTransformationRemover()
```
* Nb: avoid shadowing `str` :)
* Analogous below.

File: sumpy/qbx.py [hunk 2]
```diff
@@ -183,29 +183,30 @@ def get_kernel(self):
         raise NotImplementedError
 
     def get_optimized_kernel(self,
-            targets_is_obj_array, sources_is_obj_array, centers_is_obj_array):
+            targets_is_obj_array, sources_is_obj_array, centers_is_obj_array,
+            itgt_name="itgt"):
```
Comment (inducer) [inline sumpy/qbx.py:187]: For my education: where does `itgt` need to be overridden?

**Review by inducer — COMMENTED** [review]
LGTM, just two minor things.

File: sumpy/qbx.py (hunk 2 above)
Comment (isuruf) [reply sumpy/qbx.py:187]: It's used at https://github.com/inducer/pytential/pull/75/files#diff-4fcf38e9154a831bf71a50806d418d2ce5fbdd57cdf44c979ffc90f919e03b1dR123-R127

Comment (inducer) [reply sumpy/qbx.py:187]:
```suggestion
            # Used by pytential to override the name of the loop to be
            # parallelized. In the case of QBX, that's the loop over QBX
            # targets (not global targets).
            itgt_name="itgt"):
```

Comment (inducer) [reply sumpy/qbx.py:190]: FWIW, it took me about 20 minutes of puzzling to convince myself that what you're doing here is correct, and I'm not sold the rename makes things clearer... to be fair, I'm not claiming that the code was clear or beautiful to start with.

**Comment by inducer** [comment]
Looks OK, thanks.

**Pull request merged**